src/relationUtils.ts

```diff
@@ -276,3 +280,117 @@ export function toJson(output: Relation[] | ArrowRelation[]): any {
 
   return result;
 }
+
+export function filterRelations(relations: ArrowRelation[], keys: string[]) {
+  const len = keys.length;
+  const keysStr = keys.join('/');
+  const filteredRelations: ArrowRelation[] = [];
+
+  relations.forEach(r => {
+    const relationKeys = getKeys(r.relationId);
+
+    if (relationKeys.slice(0, len).join('/') === keysStr) {
+      const newKeys = relationKeys.slice(len);
+
+      filteredRelations.push({
+        ...r,
+        relationId: getRelationId(newKeys),
+      });
+    }
+  });
+
+  return filteredRelations;
+}
+
+export type DiagnosticPos = {
+  line: number;
+  character: number;
+};
+
+export type DiagnosticRange = {
+  start: DiagnosticPos;
+  end: DiagnosticPos;
+};
```

Comment:
these types will also be used in IC violations, so might be worth naming a bit more generically

Reply:
yep, makes sense